src/Microsoft.AspNetCore.Hosting/Internal/WebHost.cs

```diff
@@ -296,7 +296,10 @@ private void EnsureServer()
             }
 
             // Fire the IHostedService.Stop
-            _hostedServiceExecutor?.Stop();
+            if (_hostedServiceExecutor != null)
```

Comment:
Because `await _hostedServiceExecutor?.StopAsync(cancellationToken)` didn't work. I'll see if I can make this look a little better.